packages/api-client-core/spec/default-selection.spec.ts

```diff
@@ -1,11 +1,90 @@
-import type { AssertTrue, IsExact } from "conditional-type-checks";
-import type { DefaultSelection } from "../src/types.js";
-import type { AvailableTestSchemaSelection } from "./TestSchema.js";
+import type { AssertTrue, Has, IsExact } from "conditional-type-checks";
+import type { FieldSelection } from "src/FieldSelection.js";
+import type { ComputedViewFunctionWithVariables, ComputedViewFunctionWithoutVariables } from "src/GadgetFunctions.js";
+import type { AllFieldsSelected, AvailableSelection2, DefaultSelection, DefaultSelection2 } from "../src/types.js";
+import type { AvailableTestSchemaSelection, TestSchema } from "./TestSchema.js";
+import type { ExpandRecursively } from "./helpers.js";
 
-type _NullDefault = DefaultSelection<AvailableTestSchemaSelection, { select: null }, { num: true }>;
-type _TestDefaultsNullToTheDefault = AssertTrue<IsExact<_NullDefault, { num: true }>>;
+test("DefaultSelection", () => {
+  type nullDefault = DefaultSelection<AvailableTestSchemaSelection, { select: null }, { num: true }>;
+  type _TestDefaultsNullToTheDefault = AssertTrue<IsExact<nullDefault, { num: true }>>;
 
-type _NonNullDefault = DefaultSelection<AvailableTestSchemaSelection, { select: { num: false; str: true } }, { num: true }>;
-type _TestRespectsTruthySelections = AssertTrue<IsExact<_NonNullDefault, { num: false; str: true }>>;
+  type undefinedDefault = DefaultSelection<AvailableTestSchemaSelection, { select: undefined }, { num: true }>;
+  type _TestDefaultsUndefinedToTheDefault = AssertTrue<IsExact<undefinedDefault, { num: true }>>;
 
-test("true", () => undefined);
+  // eslint-disable-next-line @typescript-eslint/ban-types
+  type undefinedDefault2 = DefaultSelection<AvailableTestSchemaSelection, {}, { num: true }>;
+  type _TestDefaultsUndefinedToTheDefault2 = AssertTrue<IsExact<undefinedDefault2, { num: true }>>;
+
+  type anyDefault = DefaultSelection2<AvailableTestSchemaSelection, { select: any }, { num: true }>;
+  type _TestDefaultsAnyToTheDefault = AssertTrue<IsExact<anyDefault, { num: true }>>;
+
+  type nonDefault = DefaultSelection<AvailableTestSchemaSelection, { select: { num: false; str: true } }, { num: true }>;
+  type _TestRespectsTruthySelections = AssertTrue<IsExact<nonDefault, { num: false; str: true }>>;
+
+  type nonDefault2 = DefaultSelection<AvailableTestSchemaSelection, { select: { num: false } }, { num: true }>;
+  type _TestRespectsTruthySelections2 = AssertTrue<IsExact<nonDefault2, { num: false }>>;
+});
+
+test("AvailableSelection", () => {
+  type availableTestSchemaSelection = AvailableSelection2<TestSchema>;
+  type _TestAvailableTestSchemaSelection = Has<FieldSelection, availableTestSchemaSelection>;
+
+  type testType = { a: number; b: { c: string; d: { e: boolean } } };
+  type availableSelection = AvailableSelection2<testType>;
```

Comment:
The original AvailableSelection has a bug where boolean fields seem to end up being just `boolean | undefined` instead of `boolean | null | undefined`.